app/controllers/redaction/news_controller.rb

```diff
@@ -54,6 +58,22 @@ def reassign
     redirect_to [namespace, @news], notice: "L'auteur initial de la dépêche a été changé"
   end
 
+  def update_figure
+    params.require(:news).require([:figure_alternative, :figure_caption])
+    if params[:news][:figure_image]
+      @news.figure_image = params[:news][:figure_image]
+    end
+    else if not @news.figure_image?
```

Comment:
Ça me paraît bizarre l'enchainement `if ... end else if ... end`. Tu es sûr que le premier `end` n'est pas en trop ?

Reply:
Bien vu, c'est un oubli après plusieurs itérations: j'ai pensé au cas où l'on voulait corriger les méta-données de l'image par après. Je n'ai pas été attentif au `end` laissé en trop quand j'ai trouvé la solution.